### ISSUE TITLE
UGP support

### ISSUE THREAD

**Issue by bbondy** [brave/brave-browser#288]
We'll want to add support to Brave Core for giving users grants from our User Growth Pool (UGP)

This is a server-side configured promotion which the browser will check for and prompt the user to engage in.  Here are a few screen caps from `browser-laptop`:
![screen shot 2018-06-17 at 10 02 39 pm](https://user-images.githubusercontent.com/4733304/41519015-540fbab0-727a-11e8-8741-fb272133edb8.png)
![screen shot 2018-06-17 at 10 02 47 pm](https://user-images.githubusercontent.com/4733304/41519016-55e66cd0-727a-11e8-8c7e-522f7b45eb6b.png)

Promotion is currently checked for using the BAT client's `getPromotion` method:
https://github.com/brave-intl/bat-client/blob/9c5cb329c2e007bc8ad9a69ffa59d98d95407b37/index.js#L757

Basically the grants are fetched and have an expiration date (controlled server side). The BAT client library conveniently wraps calls to the REST service on the backend.

With `browser-laptop`, we have a captcha in place which must be solved before a user can claim the grant:
![screen shot 2018-06-17 at 10 10 02 pm](https://user-images.githubusercontent.com/4733304/41519163-3cd5c6ea-727b-11e8-9924-3d8369d76da7.png)

**Comment by LaurenWags** [comment]
@NejcZdovc is testing done for this https://github.com/brave/brave-browser/issues/928#issuecomment-426816556 sufficient to cover this issue? if not, could you suggest additional test cases? cc @kjozwiak

**Comment by NejcZdovc** [comment]
@LaurenWags yes, that should be sufficient

**Comment by LaurenWags** [comment]
Labelling as `QA/No` as this was basically tested under https://github.com/brave/brave-browser/issues/928#issuecomment-426816556 as confirmed above
